cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetsModules.java

```diff
@@ -119,7 +119,7 @@ protected void configure() {
 
   @Override
   public Module getDistributedModules() {
-    return getDistributedModules(false);
+    return getDistributedModules(true);
```

Comment:
The binding should be done by a guice Provider and the Provider can takes an injected cConf to determine which class to bind to. We shouldn't need true/false here.